### PR TITLE
update printBenchmarkResults to register task inputs

### PR DESCRIPTION
**Pull request by aSemy**
Updating Gradle config improves build performance, and prepares for future Gradle versions.

### CODE REVIEW

File: kotlin-styled-next/build.gradle.kts [hunk 1]
```diff
@@ -33,22 +35,29 @@ dependencies {
...
+    val fileNames = listOf(
+        "AddDuplicateCss",
+        "AddStyledElements",
+        "ConvertToStyled",
+        "CssBuildersInject",
+        "DataTypeOperations",
+        "InjectCssNPlusOne",
+    )
+    val reports = objects.fileCollection().from(fileNames.map { test ->
+        layout.buildDirectory.file("reports/tests/test/classes/benchmark.$test.html")
```
Comment (aSemy) [inline kotlin-styled-next/build.gradle.kts:50]: I'm not sure this directory is correct. When I run tests, I can see two, but they're `jsTest` and `jsBrowserTest`

* `kotlin-styled-next/build/reports/tests/jsBrowserTest/`
* `kotlin-styled-next/build/reports/tests/jsTest/`

Do you know if it should really be `reports/tests/test/`?

**Comment by aSemy** [comment]
I also noticed something else while looking at this code.

https://github.com/JetBrains/kotlin-wrappers/blob/299d251ea37debe07903e5d8328caa29a11cf66f/kotlin-styled-next/build.gradle.kts#L85-L91

I don't think `enabled = project.findProperty("test") == true` does anything. I can see the tests do run on CI, when `-Ptest=false`. Is the flag still used? Maybe it should be removed?

File: kotlin-styled-next/build.gradle.kts (hunk 1 above)
Comment (turansky) [reply kotlin-styled-next/build.gradle.kts:50]: Probably folder changed after migration on `multiplatform` plugin.

cc @Leonya

**Comment by turansky** [comment]
> I also noticed something else while looking at this code.
> 
> https://github.com/JetBrains/kotlin-wrappers/blob/299d251ea37debe07903e5d8328caa29a11cf66f/kotlin-styled-next/build.gradle.kts#L85-L91
> 
> I don't think `enabled = project.findProperty("test") == true` does anything. I can see the tests do run on CI, when `-Ptest=false`. Is the flag still used? Maybe it should be removed?

Yes, looks like it doesn't work at all :(

**Pull request merged**